smarts/core/opendrive_road_network.py

```diff
@@ -1248,11 +1241,13 @@ def lanes(self) -> List[RoadMap.Lane]:
         @property
         def bounding_box(self) -> Optional[BoundingBox]:
             """Get the minimal axis aligned bounding box that contains all road geometry."""
+            # XXX: This shouldn't be public.
             # XXX: The return here should be Optional[BoundingBox]
             return self._bounding_box
```

Comment:
I disagree (obviously, hence the comment I added).  In my opinion, the only public methods should be the ones specified in the `RoadMap` base class.  For example, no code elsewhere in SMARTS should ever use this method because it doesn't exist in `RoadMap` nor other map formats (and code in SMARTS should never assume a specific map format).   Put another way, SMARTS code should only deal with generic `RoadMap` objects.  

So having extra public methods like this in only one implementation of `RoadMap` can be confusing and invite problems (i.e.,  if someone tries to use this outside of `opendrive_road_network.py`).